assets/app/scripts/directives/fromFile.js

```diff
@@ -0,0 +1,316 @@
+          } else if (!item.metadata) {
+            missingField = "'metadata'";
+          }
+          $scope.error = {
+            message: "Resource is missing " + missingField + " field."
+          };
+        }
+
+        function openTemplateProcessModal() {
+          var modalInstance = $uibModal.open({
+            animation: true,
+            templateUrl: 'views/modals/process-template.html',
+            controller: 'ProcessTemplateModalController',
+            scope: $scope
+          });
+          modalInstance.result.then(function() {
+            if ($scope.templateOptions.add) {
+              createAndUpdate();
+            } else {
+              TemplateService.setTemplate($scope.resourceList[0]);
+              redirect();
+            }
+          });      
+        }
+
+        function confirmReplace() {
+          var modalInstance = $uibModal.open({
+            animation: true,
+            templateUrl: 'views/modals/confirm-replace.html',
+            controller: 'ConfirmReplaceModalController',
+            scope: $scope
+          });
+          modalInstance.result.then(function() {
+            createAndUpdate();
+          });
+        }
+
+        // create 
+        function createAndUpdate() {
+          var createUpdatePromises = [];
+          if ($scope.updateResources.length > 0) {
+            createUpdatePromises.push(updateResourceList());
+          }
+          if ($scope.createResources.length > 0) {
+            createUpdatePromises.push(createResourceList());
+          }
+          $q.all(createUpdatePromises).then(function() {
+            redirect();
+          });
+        }
+
+        function redirect() {
+          var subPath;
+          if ($scope.resourceKind === "Template" && $scope.templateOptions.process) {
+            subPath =  "create/fromtemplate?name=" + $scope.resourceName + "&namespace=" + encodeURIComponent($scope.projectName);
+          } else {
+            subPath = "overview";
+          }
+          $location.url("project/" + encodeURIComponent($scope.projectName) + "/" + subPath);
+        }
+
+        function checkResource(item) {
+          // Check if the resource already exists. If it does, replace it spec with the new one.
+          return DataService.get(APIService.kindToResource(item.kind), item.metadata.name, $scope.context, {errorNotification: false}).then(
+            // resource does exist
+            function(resource) {
+              resource.spec = item.spec;
+              $scope.updateResources.push(resource);
+            },
+            // resource doesn't exist
+            function() {
+              $scope.createResources.push(item);
+          });
+        }
+
+        function createResourceList(){
+          var titles = {
+            started: "Creating in project " + $scope.projectName,
+            success: "Created in project " + $scope.projectName,
+            failure: "Failed to create in project " + $scope.projectName
```

Comment:
"Creating resources in project..."
"Creating resources in project..."
"Failed to create some resources..."